sockeye/inference.py

```diff
@@ -130,14 +130,16 @@ class TranslatorInput:
     sentence_id: SentenceId
     tokens: Tokens
     factors: Optional[List[Tokens]] = None
+    source_prefix: Optional[Tokens] = None
+    source_prefix_factors: Optional[List[Tokens]] = None
     restrict_lexicon: Optional[lexicon.TopKLexicon] = None
     constraints: Optional[List[Tokens]] = None
     avoid_list: Optional[List[Tokens]] = None
     pass_through_dict: Optional[Dict] = None
 
     def __str__(self):
-        return 'TranslatorInput(%s, %s, factors=%s, constraints=%s, avoid=%s)' \
-            % (self.sentence_id, self.tokens, self.factors, self.constraints, self.avoid_list)
+        return 'TranslatorInput(%s, %s, factors=%s, source_prefix=%s, source_prefix_factors=%s, constraints=%s, avoid=%s)' \
+            % (self.sentence_id, self.tokens, self.factors, self.source_prefix, self.source_prefix_factors, self.constraints, self.avoid_list)
 
     def __len__(self):
         return len(self.tokens)
```

Comment:
Should this be reconsidered and take into account the prefix length?

Reply:
I also thought about that. However it is actually more convenient to not take into account of prefix length here. There are two reasons:
1. Please correct me: I think the length of input is used to compute the maximum length of translation output for the beam search (i.e. via the `get_max_input_output_length` function). Prefix tokens (e.g. <formal> <toGerman> <short>), however, should not be count as ``real" content tokens for that purpose in my opinion.
2. I found keeping the prefix length separately make it very convenient for chunking: We don't change anything regarding to the logic of chunking as we still chunk the input as is. What we need to change is simply the chunk size, which is the normal chunk size minus the prefix length.

Let me know if you have a different opinion.

Reply:
Makes sense, thanks for elaborating on it!

Reply:
One thing to think about is that prefixes are added differently during training and inference.

For training, users can add prefix tokens directly to the data because there's no chunking (sequences over max length are filtered out).  In this case, prefix tokens will be counted when computing length ratios.

During inference, the model treats each chunk as a separate sequence.  Including prefix tokens when computing the length of each chunk would match the training conditions.

Reply:
Thanks for suggestions. Since prefix tokens are counted when computing length ratios during training, let us keep it be counted at test time by adding prefix length here. Yeah I agreed.